helpers.php

```diff
@@ -1065,6 +1255,85 @@ function refundOrReleaseTransactionByStatus($transaction)
     $api->call();
 }
 
+/**
+ * Get the terminal ID based on the remote name and shop ID.
+ *
+ * @param string $remote_name
+ * @param int $shop_id
+ *
+ * @return int|string
+ */
+function getTerminalIdByRemoteName($remote_name, $shop_id = 1)
+{
+    $terminal_id = '';
+    try {
+        if (filter_var($shop_id, FILTER_VALIDATE_INT, ['options' => ['min_range' => 1]]) === false) {
+            throw new Exception('Invalid shop id');
+        }
+        $query = 'SELECT id_terminal FROM `' . _DB_PREFIX_ . "altapay_terminals` WHERE active = 1 AND remote_name = '" . pSQL($remote_name) . "' AND shop_id = '" . $shop_id . "' ";
+
+        $result = Db::getInstance()->getRow($query);
+        $terminal_id = $result['id_terminal'];
+    } catch (Exception $e) {
+        $context = Context::getContext();
+        if (isset($context->controller) && isset($context->controller->errors)) {
+            $context->controller->errors[] = $e->getMessage();
+        }
+        PrestaShopLogger::addLog($e->getMessage(), 4);
+    }
+
+    return $terminal_id;
+}
+
+/**
+ * Save transaction data to the database.
+ *
+ * @param array $result
+ * @param string $payment_form_url
+ * @param int $cartId
+ * @param string $terminalName
+ *
+ * @return void
+ */
+function saveTransactionData($result, $payment_form_url, $cartId, $terminalName)
+{
+    $sql = 'INSERT INTO `' . _DB_PREFIX_ . 'altapay_transaction`
+				(id_cart, payment_form_url, unique_id, amount, terminal_name, date_add, token) VALUES ' .
+        "('" . (int) $cartId . "', '" . pSQL($payment_form_url) . "', '" . pSQL($result['uniqueid']) . "', '"
```

Comment:
Same comments as above.

Reply:
We have added a validation for non-zero positive integer ✅